pontoon/base/models.py

```diff
@@ -377,6 +357,87 @@ def add_translation(self, string, locale, user):
 
         translation.save()
 
+    def get_translation(self, plural_form=None):
+        """Get fetched translation of a given entity."""
+        translations = self.fetched_translations
+
+        if translations:
+            if plural_form:
+                translations = [t for t in translations if t.plural_form == plural_form]
+
+            translation = sorted(translations, key=lambda k: (-k.approved, k.date))[0]
+
+            return {
+                'fuzzy': translation.fuzzy,
+                'string': translation.string,
+                'approved': translation.approved,
+                'pk': translation.pk
+            }
+
+        else:
+            return {
+                'fuzzy': False,
+                'string': u'',
+                'approved': False,
+                'pk': None
+            }
+
+    @classmethod
+    def serialize(self, project, locale, paths=None):
+        """Get project entities with locale translations."""
+
+        entities = self.objects.filter(resource__project=project, obsolete=False)
+
+        if paths:
+            entities = entities.filter(resource__path__in=paths)
+
+        entities = (entities
+                    .prefetch_related(
+                        Prefetch(
+                            'translation_set',
+                            queryset=Translation.objects.filter(locale=locale),
+                            to_attr='fetched_translations'
+                        )
+                    )
+                    .prefetch_related('resource')
+                    )
+
+        entities_array = []
+
+        for entity in entities:
+
+            try:
+                source = eval(entity.source)
+            except SyntaxError:
+                source = entity.source
+
+            translation_array = []
+
+            if entity.string_plural == "":
+                translation_array.append(entity.get_translation())
+
+            else:
+                for plural_form in range(0, locale.nplurals or 1):
+                    translation_array.append(entity.get_translation(plural_form))
+
+            entities_array.append({
+                'pk': entity.pk,
+                'original': entity.string,
+                'marked': entity.marked,
+                'original_plural': entity.string_plural,
+                'marked_plural': entity.marked_plural,
+                'key': entity.key,
+                'path': entity.resource.path,
+                'format': entity.resource.format,
+                'comment': entity.comment,
+                'order': entity.order,
+                'source': source,
+                'obsolete': entity.obsolete,
+                'translation': translation_array,
+            })
+
+        return sorted(entities_array, key=lambda k: k['order'])
```

Comment:
Alternatively, you can just sort the entities when you query them with `.order('order')` and then use `yield` to turn this method into a generator. Would save on building the list then appending.